### PR TITLE
fix: autogenerate docs again + add storage along side other resources

### PR DESCRIPTION
**Pull request by AidanAbd**

### CODE REVIEW

File: docs/source/index.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ Lacth SDK allows developers to upload workflows to the full-featured [Latch Plat
 
 With Latch SDK, developers can write the description to their workflow and customize input parameters using plain Markdown. Latch automatically parses the written text and Python function headers to compile a type-safe UI.
 
-**Specifying arbitrary cloud compute and storage resources for bioinformatics pipelines is difficult.** With Latch SDK, there are several Python task decorators that easily allow you to define the resources available at runtime. The framework starts at 2 CPUs and 4 GBs of memory and goes all the way to 31 CPUs, 120 GBs of memory and 1 GPU (24 GBs of VRAM, 9,216 CUDA cores) to easily handle all processing needs.
+**Specifying arbitrary cloud compute and storage resources for bioinformatics pipelines is difficult.** With Latch SDK, there are several Python task decorators that easily allow you to define the resources available at runtime. The framework starts at 2 CPUs and 4 GBs of memory and goes all the way to 95 CPUs, 490 GBs of memory, 5TB of storage, and 1 GPU (24 GBs of VRAM, 9,216 CUDA cores) to easily handle all processing needs.
```
Comment (ayushkamat) [inline docs/source/index.md:27]:
```suggestion
**Specifying arbitrary cloud compute and storage resources for bioinformatics pipelines is difficult.** With the Latch SDK, there are several Python task decorators that easily allow you to define the resources available at runtime. The framework starts at 2 CPUs and 4 GiBs of memory and goes all the way to 95 CPUs, 490 GBs of memory, 5TB of storage, and 1 GPU (24 GBs of VRAM, 9,216 CUDA cores) to easily handle all processing needs.
```
is it `5TB` or `5TiB`? feel like its the latter but idk

File: docs/source/getting_started/authoring_your_workflow.md [hunk 2]
```diff
@@ -126,7 +126,7 @@ To arbitrarily specify resource requirements, use:
 ```python
 from latch import custom_task
 
-@custom_task(cpu, memory)
+@custom_task(cpu, memory, storage_gib = storage_gib)
```
Comment (ayushkamat) [inline docs/source/getting_started/authoring_your_workflow.md:129]:
```suggestion
@custom_task(cpu=cpu, memory=memory, storage_gib=storage_gib)
```
for consistency lol

File: docs/source/index.md (hunk 1 above)
Comment (ayushkamat) [reply docs/source/index.md:27]: LMAO its neither actually

**Review by ayushkamat — CHANGES_REQUESTED** [review]

File: docs/source/basics/defining_cloud_resources.md [hunk 3]
```diff
@@ -5,21 +5,36 @@ the task are provisioned automatically and managed for the user until task
...
 
 The Latch SDK currently supports a set of prespecified task resource requests
 represented as decorators:
 
-* `small_task`: 2 cpus, 4 gigs of memory, 0 gpus
-* `medium_task`: 32 cpus, 128 gigs of memory, 0 gpus
-* `large_task`: 96 cpus, 192 gig sof memory, 0 gpus
-* `small_gpu_task`: 8 cpus, 32 gigs of memory, 1 gpu (24 gigs of VRAM, 9,216 cuda cores)
-* `large_gpu_task`: 31 cpus, 120 gigs of memory, 1 gpu (24 gigs of VRAM, 9,216 cuda cores)
+* `small_task`: 2 cpus, 4 gigs of memory, 500 GiB of storage, 0 gpus
```
Comment (maximsmol) [inline docs/source/basics/defining_cloud_resources.md:33]: use a table?

File: docs/source/basics/defining_cloud_resources.md [hunk 4]
```diff
@@ -5,21 +5,36 @@ the task are provisioned automatically and managed for the user until task
 completion.
 
 Tasks can be annotated with the resources they are expected to consume (eg. CPU,
-RAM, GPU) at runtime and these requests will be fullfilled during the scheduling
+RAM, Storage, GPU) at runtime and these requests will be fullfilled during the scheduling
 process.
 
+## CPU
+The number of cpu cores available to the task. If your task uses more than the alotted
```
Comment (maximsmol) [inline docs/source/basics/defining_cloud_resources.md:12]: "your task" -> "the task"

File: docs/source/basics/defining_cloud_resources.md [hunk 5]
```diff
@@ -5,21 +5,36 @@ the task are provisioned automatically and managed for the user until task
...
+## Memory
+The amount of memory available to the task. If your task uses more than the alotted
+amount of memory, the task will crash.
+
+## Storage
+The amount of disc space available to the task. If your task uses more than the alotted
+amount of storage, the task will crash.
+
+## GPU
+The number of GPUs available to the task.
```
Comment (maximsmol) [inline docs/source/basics/defining_cloud_resources.md:24]: explain that running out of VRAM will usually trigger an exception and that using more compute power than available will usually just mean you run slowly

File: docs/source/basics/defining_cloud_resources.md [hunk 6]
```diff
@@ -41,11 +56,11 @@ def inference(
 
 ## Custom Task Resource
 
-You can also arbitrarily specify task resources using `@custom_task`:
+You can also specify task resources using `@custom_task`:
```
Comment (maximsmol) [inline docs/source/basics/defining_cloud_resources.md:59]: give min/max values here, explain different node groups and what the tresholds are, link to the API doc

File: docs/source/basics/defining_cloud_resources.md [hunk 7]
```diff
@@ -41,11 +56,11 @@ def inference(
...
 ```python
 from latch import custom_task
 
-@custom_task(cpu, memory) # cpu: int, memory: int
+@custom_task(cpu, memory, storage_gib = storage_gib) # cpu: int, memory: int, storage_gib: int
```
Comment (maximsmol) [inline docs/source/basics/defining_cloud_resources.md:63]: del the comment, API docs have types

File: docs/source/getting_started/authoring_your_workflow.md (hunk 2 above)
Comment (maximsmol) [inline docs/source/getting_started/authoring_your_workflow.md:129]: examples should use concrete values e.g. 
```suggestion
@custom_task(cpu=8, memory=64, storage_gib=1000)
```

File: docs/source/getting_started/authoring_your_workflow.md [hunk 8]
```diff
@@ -107,13 +107,13 @@ Specifying compute and storage requirements is as easy as using a Python decorat
...
 
-@small_task # 2 cpus, 4 gigs of memory, 0 gpus
+@small_task # 2 cpus, 4 gigs of memory, 500 GiB of storage, 0 gpus
 def my_task(
     ...
 ):
     ...
 
-@large_gpu_task #31 cpus, 120 gigs of memory, 1 gpu
+@large_gpu_task #31 cpus, 120 gigs of memory, 2000 GiB of storage, 1 gpu
```
Comment (maximsmol) [inline docs/source/getting_started/authoring_your_workflow.md:116]:
```suggestion
@large_gpu_task # 31 cpus, 120 gigs of memory, 2000 GiB of storage, 1 gpu
```

File: docs/source/getting_started/authoring_your_workflow.md (hunk 2 above)
Comment (maximsmol) [reply docs/source/getting_started/authoring_your_workflow.md:129]: also black code style does not leave spaces around `=`

File: docs/source/index.md [hunk 9]
```diff
@@ -24,7 +24,7 @@ Lacth SDK allows developers to upload workflows to the full-featured [Latch Plat
 
 With Latch SDK, developers can write the description to their workflow and customize input parameters using plain Markdown. Latch automatically parses the written text and Python function headers to compile a type-safe UI.
 
-**Specifying arbitrary cloud compute and storage resources for bioinformatics pipelines is difficult.** With Latch SDK, there are several Python task decorators that easily allow you to define the resources available at runtime. The framework starts at 2 CPUs and 4 GBs of memory and goes all the way to 31 CPUs, 120 GBs of memory and 1 GPU (24 GBs of VRAM, 9,216 CUDA cores) to easily handle all processing needs.
+**Specifying arbitrary cloud compute and storage resources for bioinformatics pipelines is difficult.** With the Latch SDK, there are several Python task decorators that easily allow you to define the resources available at runtime. The framework starts at 2 CPUs and 4 GiBs of memory and goes all the way to 95 CPUs, 490 GBs of memory, 4949 GiB of storage, and 1 GPU (24 GBs of VRAM, 9,216 CUDA cores) to easily handle all processing needs.
```
Comment (maximsmol) [inline docs/source/index.md:27]: note that it's not all at the same time (i.e. CPU limit for big-ram nodes is lower)

"GB" -> "GiB"

**Review by maximsmol — CHANGES_REQUESTED** [review]

**Pull request merged**